osquery/core/watcher.cpp

```diff
@@ -339,26 +359,18 @@ void WatcherRunner::createWorker() {
     return;
   }
 
-  auto worker_pid = fork();
-  if (worker_pid < 0) {
+  auto worker = PlatformProcess::launchWorker(exec_path.string(), argv_[0]);
+  if (worker.get() == nullptr) {
```

Comment:
Do you need to use `.get()` here? I believe a `std::shared_ptr` can compare safely with `nullptr`.